torch/_refs/__init__.py

```diff
@@ -3614,6 +3614,23 @@ def repeat(a: Tensor, *repeat_shape) -> Tensor:
     return permuted_result.reshape(target_shape)
 
 
+def _nd_to_1d(a: Tensor) -> Tensor:
+    torch._check(a.numel() > 0)
+    torch._check(a.ndim > 0)
+    torch._check(a.is_contiguous())
+    return torch.as_strided(a, [a.numel()], [1])
+
+
+def _1d_to_2d(a: Tensor, dim0, dim1) -> Tensor:
+    torch._check(a.numel() > 0)
+    torch._check(a.ndim == 1)
+    torch._check(a.is_contiguous())
+    torch._check_is_size(dim0)
+    torch._check_is_size(dim1)
```

Comment:
This is not quite right. You need to account for when dim0 or dim1 == -1

Reply:
I believe the inference is being done in ```_reshape_view_helper``` before the call to these functions. See line 3638.

I am going to add comments to these functions.

Reply:
oh oops, you're right. But then do you even need these asserts?

Reply:
I wrote them more as "preconditions" i.e., facts which must be guaranteed by preceding code. I don't think I need them for solving Ivan's problem. I will remove them.